be2-scala/src/test/scala/ch/epfl/pop/decentralized/MonitorSuite.scala

```diff
@@ -99,4 +104,40 @@ class MonitorSuite extends TestKit(ActorSystem("MonitorSuiteActorSystem")) with
     testProbe.send(monitorRef, Monitor.AtLeastOneServerConnected)
     testProbe.expectMsgType[Monitor.GenerateAndSendHeartbeat](timeout)
   }
+
+  test("monitor should send ConnectTo() requests to ConnectionMediator upon relevant config file change") {
+    val mockConnectionMediator = TestProbe()
+    val monitorRef = system.actorOf(Monitor.props(ActorRef.noSender))
+
+    // Ping monitor to inform it of ConnectionMediatorRef
+    mockConnectionMediator.send(monitorRef, ConnectionMediator.Ping())
+
+    // Expect no message as long as the server peers list is untouched
+    mockConnectionMediator.expectNoMessage(timeout)
+
+    val newContent = List("some", "strings")
+    testWriteToServerPeersConfig(newContent)
+
+    mockConnectionMediator.expectMsgType[ConnectionMediator.ConnectTo](timeout)
+    deleteTestConfig()
```

Comment:
I don't think the test config file will be deleted if the test failed in the previous assertion

Reply:
Fixed !